### PR TITLE
Fix transfer window deadlock with Qt, implement Window::doneCurrent()

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
Hold on with merge. The tfer window creation in tfer thread does not work with WGL.

File: eq/pipe.cpp [hunk 1]
```diff
@@ -136,10 +137,17 @@ class TransferThread : public co::Worker
...
+    {
+        LB_TS_THREAD( _pipe->_tferThread );
+        co::Worker::run();
+    }
+
     bool stopRunning() override { return _stop; }
     void postStop() { _stop = true; }
 
 private:
+    const eq::Pipe* _pipe;
```
Comment (eile) [inline eq/pipe.cpp]: _pipe still needed?

File: eq/qt/windowSystem.cpp [hunk 2]
```diff
@@ -65,14 +66,32 @@ eq::SystemWindow* WindowSystem::createWindow( eq::Window* window,
...
-
-    QCoreApplication* app = QApplication::instance();
-    app->connect( qtWindow, SIGNAL( destroyImpl( detail::Window* )),
-                  _factory, SLOT( onDestroyImpl( detail::Window* )));
 
+    if( settings.getIAttribute(
+                WindowSettings::IATTR_HINT_DRAWABLE ) != eq::OFF )
+    {
+        // QWindow creation/destruction must happen in the app thread;
+        // unblock main thread to give QApplication the change to process the
```
Comment (eile) [inline eq/qt/windowSystem.cpp]: s/change/chance/

File: eq/pipe.cpp (hunk 1 above)
Comment (tribal-tec) [inline eq/pipe.cpp]: in run(): LB_TS_THREAD( _pipe->_tferThread );

**Comment by tribal-tec** [comment]
Just did some archaeology: I should not introduce bug #177 again. Shared context creation must remain in the render thread.

**Comment by tribal-tec** [comment]
Updated

**Comment by tribal-tec** [comment]
retest this please

**Comment by tribal-tec** [comment]
retest this please

File: eq/node.cpp [hunk 3]
```diff
@@ -331,7 +331,7 @@ void Node::_finishFrame( const uint32_t frameNumber ) const
     const Pipes& pipes = getPipes();
     for( Pipes::const_iterator i = pipes.begin(); i != pipes.end(); ++i )
     {
-        const Pipe* pipe = *i;
```
Comment (eile) [inline eq/node.cpp]: why removed?

**Comment by eile** [comment]
+1

**Pull request merged**